### ISSUE TITLE
Make sharing the trace work

### ISSUE THREAD

**Issue by tahini** [theia-ide/theia-trace-extension#19]
Next to the trace name in the trace explorer is a button to copy-paste a URL to share with someone else. The source URL does not work right now

**Comment by marcdumais-work** [comment]
related to: https://github.com/theia-ide/theia-trace-extension/issues/79

Just food for thoughts: 

Sharing features are often implemented one level above, in a workspace server (Gitpod, Che). I think the reason might be that [checkpoint](https://docs.docker.com/engine/reference/commandline/checkpoint/)-ing a docker container is relatively easy, and can be automated for such scenarios. Then anyone that accesses the shared workspace would be served a fresh copy of the checkpoint, that includes the running state of the container. This has many advantages since you can share not only a trace, but a working, running workspace, with test app potentially pre-built and ready for showtime. We think we can eventually restore e.g. a debug session too, the problem's traces pre-generated and pre-loaded in the tracing app, etc.

---

Now, imagine: when you use your workspace server as part of your CI, like we will Gitpod soon, the CI runs in the same type of workspace container that would be served for interactive use, and it would be possible run the tests through the debugger, with e.g. un-handled exception breakpoints enabled. When such a BP is hit, we trigger a snapshot and e.g. tag and share with the sheriff or last person who touched that code. Or maybe open an issue about the problem and adds "live" workspace, with the problem happening in a suspended debug session. .

The idea is not new, nor mine. @MatthewKhouzam might be #PatientZero or at least a transmitter. But it's the first time I see how it can work more concretely. 

TL;DR Assuming that using Gitpod is ok with the main contributors and their employers, we could consider using its workspace sharing feature(s) to get the bronze level of the above. Needs further investigation.
